db/change_cache.go

```diff
@@ -827,12 +848,15 @@ func (h *LogPriorityQueue) Pop() interface{} {
 //////// SKIPPED SEQUENCE QUEUE
 
 func (c *changeCache) RemoveSkipped(x uint64) error {
+	c.context.DbStats.statsCacheMap.Add(base.StatKeySkippedSeqLen, -1)
```

Comment:
All of these stats are being incremented/decremented, but I think it might be safer to instead always be setting the current length.

There are some cases right now where we change the stat, then run the operation (which could error) and produce invalid length stats.